master/internal/stream/system.go

```diff
@@ -467,6 +485,13 @@ func prepareMessageWithCache(
 	return *cache
 }
 
+func newDeletedInterface(key, deleted string, deleteFunc stream.DeleteFunc) interface{} {
+	if deleteFunc == nil {
```

Comment:
nit: DeleteMsg() and UpsertMsg() both check following this pattern:
```
if <func> != nil {
    return <func>
} 
return <default behavior>
```
Is there a reason we're doing the inverse here?